setup.py

```diff
@@ -17,11 +19,194 @@
 #  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 #  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 #  DEALINGS IN THE SOFTWARE.
+"""Molecule distribution package setuptools installer."""
 
 import setuptools
+import setuptools.command.dist_info
 
-setuptools.setup(
-    packages=setuptools.find_packages(),
-    pbr=True,
-    setup_requires=['pbr']
-)
+
+ALL_STRING_TYPES = tuple(map(type, ('', b'', u'')))
```

Comment:
Just to note, this would all be very scary and signal some major maintenance headaches but it seems to be doing just fine over at https://github.com/ansible/ansible-lint/blob/master/setup.py! So, please take that into consideration when reviewing. Maybe this shim itself needs to be packaged up ... ;)

Reply:
Nope, you cannot package this shim. Just because of a chicken/egg problem :)
Of course, you could do even more nasty stuff but I urge you not to.

The right thing to do is to actually upgrade pip/setuptools/python to something better maintainable and if future you can even expect something as magical as https://twitter.com/webKnjaZ/status/1113530472808763394

Reply:
** But I was thinking about making a repo with an example like this...

Reply:
Oh, and this version is way more improved over the one used in ansible-lint...